wardrobe/tradable_kitties/src/lib.rs

```diff
@@ -0,0 +1,573 @@
+    FreeBreedingCannotBeUpdated,
+    /// Kitty NumOfBreeding cannot be updated.
+    NumOfBreedingCannotBeUpdated,
+    /// Updated price of is incorrect.
+    UpdatedKittyIncorrectPrice,
+
+    /// No input for kitty buy operation.
+    InputMissingBuyingNothing,
+    /// No output for kitty buy operation.
+    OutputMissingBuyingNothing,
+    /// Incorrect number of outputs buy operation.
+    IncorrectNumberOfInputKittiesForBuyOperation,
+    /// Incorrect number of outputs for buy operation.
+    IncorrectNumberOfOutputKittiesForBuyOperation,
+    /// Kitty not avilable for sale
+    KittyNotForSale,
+    /// Kitty price cant be none when it is avilable for sale
+    KittyPriceCantBeNone,
+    /// Kitty price cant be zero when it is avilable for sale
+    KittyPriceCantBeZero,
+    /// Not enough amount to buy kitty
+    InsufficientCollateralToBuyKitty,
+
+    // From below money constraintchecker errors are added
+    /// The transaction attempts to spend without consuming any inputs.
+    /// Either the output value will exceed the input value, or if there are no outputs,
+    /// it is a waste of processing power, so it is not allowed.
+    SpendingNothing,
+    /// The value of the spent input coins is less than the value of the newly created
+    /// output coins. This would lead to money creation and is not allowed.
+    OutputsExceedInputs,
+    /// The value consumed or created by this transaction overflows the value type.
+    /// This could lead to problems like https://bitcointalk.org/index.php?topic=823.0
+    ValueOverflow,
+    /// The transaction attempted to create a coin with zero value. This is not allowed
+    /// because it wastes state space.
+    ZeroValueCoin,
+}
+
+impl From<money::ConstraintCheckerError> for TradableKittyConstraintCheckerError {
+    fn from(error: money::ConstraintCheckerError) -> Self {
+        match error {
+            money::ConstraintCheckerError::BadlyTyped => {
+                TradableKittyConstraintCheckerError::BadlyTyped
+            }
+            money::ConstraintCheckerError::MintingWithInputs => {
+                TradableKittyConstraintCheckerError::MintingWithInputs
+            }
+            money::ConstraintCheckerError::MintingNothing => {
+                TradableKittyConstraintCheckerError::MintingNothing
+            }
+            money::ConstraintCheckerError::SpendingNothing => {
+                TradableKittyConstraintCheckerError::SpendingNothing
+            }
+            money::ConstraintCheckerError::OutputsExceedInputs => {
+                TradableKittyConstraintCheckerError::OutputsExceedInputs
+            }
+            money::ConstraintCheckerError::ValueOverflow => {
+                TradableKittyConstraintCheckerError::ValueOverflow
+            }
+            money::ConstraintCheckerError::ZeroValueCoin => {
+                TradableKittyConstraintCheckerError::ZeroValueCoin
+            }
+        }
+    }
+}
```

Comment:
This will be way simpler if you restructure your error type as recommended above.